universal-profiling-integration/src/main/java/co/elastic/otel/UniversalProfilingProcessor.java

```diff
@@ -200,10 +215,12 @@ public boolean isEndRequired() {
   @Nullable
   private void onContextChange(@Nullable Context previous, @Nullable Context next) {
     try {
-      Span oldSpan = safeSpanFromContext(previous);
-      Span newSpan = safeSpanFromContext(next);
-      if (oldSpan != newSpan && !oldSpan.getSpanContext().equals(newSpan.getSpanContext())) {
-        ProfilerSharedMemoryWriter.updateThreadCorrelationStorage(newSpan);
+      if (tlsPropagationActive) {
```

Comment:
[nit] maybe moving this check outside of the try/catch block with an early return to remove a few curly braces.